### PR TITLE
add React/noRenderReturnValue

### PR DESCRIPTION
**Pull request by yeonjuan**
rule : https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-render-return-value.md

ref: #341

### CODE REVIEW

File: packages/@romejs/compiler/lint/rules/react/noRenderReturnValue.test.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
```
Comment (ematipico) [inline packages/@romejs/compiler/lint/rules/react/noRenderReturnValue.test.ts:6]: No need this header for new files
 
```suggestion 
```

File: packages/@romejs/compiler/lint/rules/react/noRenderReturnValue.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+/**
+* Copyright (c) Facebook, Inc. and its affiliates.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
```
Comment (ematipico) [inline packages/@romejs/compiler/lint/rules/react/noRenderReturnValue.ts:6]:
```suggestion
```

File: packages/@romejs/diagnostics/descriptions/lint.ts [hunk 3]
```diff
@@ -494,6 +494,10 @@ export const lint = createDiagnosticsCategory({
 		category: "lint/react/requireRenderReturn",
 		message: "The <emphasis>render</emphasis> method on a component must return content.",
 	},
+	REACT_NO_RENDER_RETURN_VALUE: {
+		category: "lint/react/noRenderReturnValue",
+		message: "Do not depend on the return value from <emphasis>ReactDOM.render()</emphasis>",
```
Comment (ematipico) [inline packages/@romejs/diagnostics/descriptions/lint.ts:499]:
```suggestion
		message: "Do not depend on the return value from <emphasis>ReactDOM.render()</emphasis>.",
```

File: packages/@romejs/compiler/lint/rules/react/noRenderReturnValue.test.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+					`
+					let foo;
+					foo = ReactDOM.render(<div />, document.body);
+					`,
+					`
+					function render () {
+						return ReactDOM.render(<div />, document.body)
+					}`,
+				],
+				valid: ["ReactDOM.render(<div />, document.body);"],
```
Comment (ematipico) [inline packages/@romejs/compiler/lint/rules/react/noRenderReturnValue.test.ts:35]: We could add a couple of more cases:

```js
function render() {
   ReactDOM.render(<div />, document.body)
}

// ...

const foo = () => { ReactDOM.render(<div />, document.body) };
```

**Comment by yeonjuan** [comment]
@sebmck Thanks, I changed it as your suggestions :)

**Comment by sebmck** [comment]
Awesome! Looks good! The diagnostic also now points to the specific `ReactDOM.render` call which is great. Thank you!

**Pull request merged**